install/kubernetes/templates/helm/istio/values.yaml.tmpl

```diff
@@ -3,8 +3,8 @@ Istio:
    initializer_enabled: true 
    mixer_enabled: true 
    pilot_enabled: true 
-   ca_hub: gcr.io/istio-testing
-   ca_tag: d10e0c4aa05ca726cae71aa5d033dea4f7bc26e4
+   ca_hub: {CA_HUB}
```

Comment:
Which process would render values.yaml?  Originally the idea was to update tags to the current values by running updateVersion.sh. Appreciate if you could explain the reason behind this change.

Reply:
@sbezverk  line 185 renders values.yaml.  The problem here is there is no default values.yaml in the repository, so the helm chart is broken unless someone runs updateVersion.sh.  I've heard various commentary that updateVersion.sh is being deprecated in the medium term, however, until that happens I believe it is necessary to keep these files in the repository.

This exposes a problem that we end up with a whole lot of generated code checked into the Istio repository and it is unclear how this generated code receives updates on image hash changes not directed by the developer via updateVersions.sh (IOW directed by infrastructure code).

Countering that problem is that if someone runs updateVersions.sh with custom hub or tag, their git repo is dirty with their locally modified manifests.

Reply:
Helm was a recent addition and it's the only one modify its file in-place.  This change makes helm consistent with the other existing components whereby any yaml file that gets modified by updateVersion.sh is derived from a template file (in the templates/ directory) that is used to create a new file under install/ or samples/ .  This does mean that a dev needs to run updateVersions.sh to test local/new changes, but this has already been a necessity.  The checked in generated files used to be created as part of making a release, but this hasn't happened since the 0.2 releases so the generated files I'm deleting are actually from Oct (so there's probably cases of people unintentionally using stale files).  This is partly why I've been tempted to put even static (unmodified by updateVersions.sh) under the templates directory so people are more apt to run updateVersions.sh rather than accidently try to use a set of files under install/ and samples/ that at first glance appear complete (but really lack the generated files, and will be tricky for newcomers to debug).